imap_processing/cli.py

```diff
@@ -104,37 +148,101 @@ def _validate_args(args):
+    if args.data_level not in imap_processing.PROCESSING_LEVELS[args.instrument]:
         raise ValueError(
-            f"{args.level} is not a supported data level for the {args.instrument}"
+            f"{args.data_level} is not a supported data level for the {args.instrument}"
             " instrument, valid levels are: "
             f"{imap_processing.PROCESSING_LEVELS[args.instrument]}"
         )
-    if args.data_dir:
-        data_path = Path(args.data_dir)
-        if not data_path.exists():
-            raise ValueError(f"Data directory {args.data_dir} does not exist")
-        # Set the data directory to the user-supplied value
-        imap_data_access.config["DATA_DIR"] = data_path
 
 
 class ProcessInstrument(ABC):
     """An abstract base class containing a method to process an instrument.
 
     Attributes
     ----------
-    level : str
+    data_level : str
         The data level to process (e.g. ``l1a``)
+    dependency_str : str
+        A string representation of the dependencies for the instrument in the
+        format: "[{
+            'instrument': 'mag',
+            'data_level': 'l0',
+            'descriptor': 'sci',
+            'version': 'v00-01',
+            'start_date': '20231212',
+            'end_date': '20231212'
+        }]"
+    start_date : str
+        The start date for the output data. Format: YYYYMMDD
+    end_date : str
+        The end date for the output data. Format: YYYYMMDD
+    version : str
+        The version of the data. Format: vxx-xx
+    upload_to_sdc : bool
+        A flag indicating whether to upload the output file to the SDC.
     """
 
-    def __init__(self, level, file_path):
-        self.level = level
-        self.file_path = file_path
+    def __init__(
+        self,
+        data_level: str,
+        dependency_str: str,
+        start_date: str,
+        end_date: str,
+        version: str,
+        upload_to_sdc: bool,
+    ) -> None:
+        self.data_level = data_level
+
+        # Convert string into a dictionary
+        self.dependencies = loads(dependency_str.replace("'", '"'))
+
+        self.start_date = start_date
+        self.end_date = end_date
+        if not end_date:
+            self.end_date = start_date
+            print(f"Setting end time to start time: {start_date}")
+
+        self.version = version
+        self.upload_to_sdc = upload_to_sdc
+
+    def download_dependencies(self):
+        """Download the dependencies for the instrument.
+
+        Returns
+        -------
+        file_list: list[str]
+            A list of file paths to the downloaded dependencies.
+        """
+        file_list = []
+        for dependency in self.dependencies:
+            try:
+                # TODO: Validate dep dict
+                # TODO: determine what dependency information is optional
+                # TODO: Add in timestamps and descriptor to query
+                return_query = imap_data_access.query(
```

Comment:
This would mean I can't run local processing without an internet connection/access to the SDC servers? Do we need to add some checks in the query to add a "local" command to query the local files too? This can come in follow-up issues/PRs